src/pages/index.js

```diff
@@ -16,79 +22,241 @@ const cardListEl = document.querySelector(".cards__list");
+  section = new Section(
+    {
+      items: initialCards,
+      renderer: renderCard,
+    },
+    ".cards__list"
+  );
+  section.renderItems();
+});
+
+api
+  .getUserInfo()
+  .then((userInfoData) => {
+    userInfo.setUserInfo(userInfoData);
+    userInfo.setUserProfileImage(userInfoData.avatar);
+  })
 
-section.renderItems();
+  .catch((err) => {
+    console.error(err);
+  });
+
+//instantiating other classes
 
 const popupWithImage = new PopupWithImage("#card-image");
-popupWithImage.setEventListeners();
 
 const newCardModal = new PopupWithForm(
   "#add-card-modal",
   handleAddCardFormSubmit
 );
-newCardModal.setEventListeners();
 
 const editCardModal = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
+
+const deleteCardModal = new DeleteConfirmation(
+  "#delete-card-confirmation",
+  deleteCardForm,
+  handleDeleteCard
+);
+
+const profilePictureModal = new PopupWithForm(
+  "#change-profile-picture-modal",
+  handleChangeProfilePictureSubmit
+);
+
+// set event listeners
+
+popupWithImage.setEventListeners();
+
+newCardModal.setEventListeners();
+
 editCardModal.setEventListeners();
 
-const userInfo = new UserInfo({
-  titleSelector: ".profile__title",
-  descriptionSelector: ".profile__description",
+deleteCardModal.setEventListeners();
+
+profilePictureModal.setEventListeners();
+
+//button Event Listeners
+
+profileEditButton.addEventListener("click", () => {
+  const userData = userInfo.getUserInfo();
+  profileTitleInput.value = userData.title;
+  profileDescriptionInput.value = userData.description;
+  editCardModal.open();
 });
 
-function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data);
-  editCardModal.close();
+profileImageEditButton.addEventListener("click", () => {
+  profilePictureModal.open();
+});
+
+addNewCardButton.addEventListener("click", () => newCardModal.open());
+
+//functions
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    cardSelector,
+    showPreviewImage,
+    handleLikeCard,
+    handleDeleteCardPopup
+  );
+  section.addItem(card.getView());
+}
+
+function handleAddCardFormSubmit({ title, url }) {
+  addCardModalSubmitButton.innerHTML = "Saving...";
+  api
+    .postCard({ name: title, link: url })
+    .then((cardData) => {
+      renderCard(cardData, cardListEl);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addCardFormElement.reset();
+      addCardModalSubmitButton.classList.add("modal__button_disabled");
+      newCardModal.close();
+      addCardModalSubmitButton.innerHTML = "Create";
+    });
 }
 
-function handleAddCardFormSubmit(cardData) {
-  renderCard({ name: cardData.title, link: cardData.url }, cardListEl);
-  addCardFormElement.reset();
-  newCardModal.close();
+function handleDeleteCardPopup(card) {
+  deleteCardModal.open(card);
+}
+
+function handleDeleteCard(card) {
+  api
+    .deleteCard(card.getCardId())
+    .then(() => {
+      card._element.remove();
```

Comment:
`_element`

Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.

**Please, correct it everywhere**